mesonbuild/modules/gnome.py

```diff
@@ -702,14 +702,14 @@ def _get_dependencies_flags_raw(
                         lib_dir = os.path.dirname(flag)
                         external_ldflags.update([f'-L{lib_dir}'])
                         if include_rpath:
-                            external_ldflags.update([f'-Wl,-rpath {lib_dir}'])
+                            external_ldflags.update([f'-Wl,-rpath,{lib_dir}'])
                         libname = os.path.basename(flag)
                         if libname.startswith("lib"):
                             libname = libname[3:]
                         libname = libname.split(".so")[0]
                         flag = f"-l{libname}"
                     # FIXME: Hack to avoid passing some compiler options in
-                    if flag.startswith("-W"):
+                    if flag.startswith("-W") and not flag.startswith('-Wl,-rpath'):
```

Comment:
```suggestion
                    if flag.startswith("-W") and not flag.startswith('-Wl,-rpath,'):
```

Otherwise `-Wl,-rpath`  will be allowed through without the RPATH following it.